### PR TITLE
test(basic): add a basic test suite using jest

### PR DESCRIPTION
**Pull request by devenbansod**
Signed-off-by: Deven Bansod <devenbansod.bits@gmail.com>

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            };
+        case 'yarn':
+        default:
+            return {
+                error: `Error is being thrown that ${packageManagerCommand} does not exist`
+            };
+    }
+});
+
+describe('Basic test suite', () => {
```
Comment (hemal7735) [inline test/index.test.js]: @devenbansod 
What do you think about grouping based on functionality?
Like `describe("Something for isAvailable")` and `describe("something for findAll")`

**Comment by hemal7735** [comment]
@devenbansod thank you for the PR. It looks solid. 🔥 
I was wondering if jest provides conditional spying. Like we really don't need to mock the response if it is npm, as npm comes with node by default, so it will be there always.

File: test/index.test.js (hunk 1 above)
Comment (devenbansod) [reply test/index.test.js]: Makes sense. Let me try to change it to something like you suggested. 👍

**Comment by devenbansod** [comment]
@hemal7735 I believe we would want to test out the logic of our code with our tests. I'd prefer to keep the tests independent of the underlying `child_process` module.

What do you think?

File: test/index.test.js (hunk 1 above)
Comment (devenbansod) [reply test/index.test.js]: @hemal7735 what do you think about the test-suite now? With this, I guess the sentence makes sense now: `isAvailable API should return true if package manager is available`

```
> findall-pkg-mgr@1.1.0 test /Users/debansod/personal/findall-pkg-mgr
> jest

 PASS  test/index.test.js
  isAvailable API
    ✓ should return true if package manager is available (5ms)
    ✓ should return false if package manager is NOT available (1ms)
  findAll API
    ✓ should return all available package managers (1ms)
```

**Comment by hemal7735** [comment]
> @hemal7735 I believe we would want to test out the logic of our code with our tests. I'd prefer to keep the tests independent of the underlying `child_process` module.
> 
> What do you think?

Agreed 👍

**Pull request merged**